### PR TITLE
Add NearestNDInterpolator to cupyx.scipy.interpolate

### PR DESCRIPTION
**Pull request by andfoy**
See #7186
Depends on #8208 for module ordering

### CODE REVIEW

File: cupyx/scipy/interpolate/_ndgriddata.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+
+class NearestNDInterpolator(NDInterpolatorBase):
+    """NearestNDInterpolator(x, y).
+
+    Nearest-neighbor interpolator in N > 1 dimensions.
+
+    Methods
+    -------
+    __call__
```
Comment (ev-br) [inline cupyx/scipy/interpolate/_ndgriddata.py:23]: I think the `Methods` section needs to be removed to unblock the doc build. 
This is some incompatibility of the scipy and cupy doc builds : the former tolerates both autogenerated and manually added methods section, but the latter does not.

**Comment by andfoy** [comment]
imagen

I can locally confirm that the tests are passing, maybe the displayed error is related to the version of SciPy?

**Comment by ev-br** [comment]
Is this about  `distance_upper_bound`? This is indeed new in SciPy 1.12:  https://github.com/scipy/scipy/pull/19483

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
Sorry for my late review. Thanks for your contribution. LGTM!

**Comment by andfoy** [comment]
Thanks @asi1024!

**Pull request merged**